server/src/services/commits.ts

```diff
@@ -163,7 +164,63 @@ const completeCommit = async (commitId: number) => {
     commitStatus: 'completed',
   };
 
-  return commitStorage.updateCommit(commitId, fieldsToUpdate, commit.listingId);
+  const updatedCommit = await commitStorage.updateCommit(
+    commitId,
+    fieldsToUpdate,
+    commit.listingId
+  );
+  await sendCommitMessage(updatedCommit);
+  return updatedCommit;
+};
+
+/**
+ * A helper function that sends a message to customer, notifying them that their
+ * commit status has changed to 'successful', 'unsuccessful' or 'completed'.
+ * It uses Spot Messages API.
+ * @param commit The updated commit whose new status is to be notified to customer
+ */
+const sendCommitMessage = async (commit: CommitResponse) => {
+  const listing = await listingStorage.getListing(commit.listingId);
+  const customer = await customerStorage.getCustomer(commit.customerId);
+
+  let text = '';
+  switch (commit.commitStatus) {
+    case 'successful':
+      text = 'Group Buy is successful! Please proceed to payment.';
+      break;
+    case 'unsuccessful':
+      text =
+        "Unfortunately, Group Buy is unsuccessful as there weren't enough buyers.";
+      break;
+    case 'completed':
+      text = 'Your item has been delivered by the merchant!';
+      break;
+    default:
+      throw new InternalServerError(
+        'Invalid commitStatus for sending a message.'
+      );
+  }
+
+  const completeCommitMessage = {
+    imageUrl: listing.imgUrl,
+    title: listing.name,
+    text,
+    actions: [
+      {
+        label: 'View Listing',
+        url: encodeMicroappsUrl(`listing/${listing.id}`),
+      },
+    ],
+    recipients: [
+      {
```

Comment:
Looks like there is always 1 recipient only for each completeCommitMessage? Can there be a scenario when there are multiple recipients being included in one message?

Reply:
@xlai20 Based on [Messages API docs](https://developers.google.com/pay/spot/eap/reference/messages-api#re-engagement_message), it says that "Only 1 recipient is allowed per request". It's weird that they use an array but, I don't know, maybe it's for future versions where they allow multiple recipients.